File: app/src/main/java/com/botijasoftware/utils/collision/CollisionPoint.java
package com.botijasoftware.utils.collision;

import com.botijasoftware.utils.Vector3;

public class CollisionPoint extends CollisionVolume {
	private Vector3 point;
	
	public CollisionPoint(Vector3 point) {
		this.point = point;
	}
	
	public boolean collides(CollisionVolume c) {
		return c.collides(point);
	}
	
	public boolean collides(Sphere s) {
		return s.collides(point);
	}
	
	
	public boolean collides(Capsule c) {
		return c.collides(point);
	}
	
	public boolean collides(Vector3 v) {
		return v.equals(point);
	}
	
	public Vector3 getPosition() {
		return point;
	}
	
	public void setPosition(Vector3 v) {
		point = v;
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/collision/CollisionPoint.kt
package com.botijasoftware.utils.collision

import com.botijasoftware.utils.Vector3

class CollisionPoint(override var position: Vector3) : CollisionVolume() {

    override fun collides(c: CollisionVolume): Boolean {
        return c.collides(position)
    }

    override fun collides(s: Sphere): Boolean {
        return s.collides(position)
    }


    override fun collides(c: Capsule): Boolean {
        return c.collides(position)
    }

    override fun collides(v: Vector3): Boolean {
        return v.equals(position)
    }

}